### ISSUE TITLE
Confusing solver-interface XML tag and impact on geometric multiscale

### ISSUE THREAD

**Issue by MakisH** [precice/precice#1416]
**Please describe the problem you are trying to solve.**

Currently, every preCICE configuration file starts with:

```xml
<solver-interface dimensions="..." />
```

[Everything, apart from the logger, is wrapped by this tag](https://precice.org/configuration-xml-reference.html). The reason seems to be an implementation detail, as this is the configuration of the `SolverInterface` class.

However, I see two main issues with this tag:

- **Issue 1: Confusing name "solver-interface".**
    As a "solver interface", I would imagine the (physical or code) interface between two participants. This is already ambiguous, as a new user could think: "is it the part of the mesh that belongs to the interface? is it the code coupling pair?". Besides, all coupling schemes are defined inside this tag, which could confuse people in terms of singular vs plurar. But overall, this is always awkward to explain to people: "just use this in your config".
- **Issue 2: Global dimensions.**
  Moved to https://github.com/precice/precice/issues/1466
  This issue was originally more about the dimensions from my side, but the discussion focused on the more fundamental issue of the tag itself.

**Describe the solution you propose.**

We could consider alternative names for the tag.

**Describe alternatives you've considered**

Removing the tag altogether and bringing everything to the top level would mean that we need to find a new place not only for the `dimensions` attribute, but also for the `experimental` flag. It would also make an extension to support more `SolverInterface` objects in a participant more complicated (see https://github.com/precice/precice/issues/655).

**Comment by uekerman** [comment]
The technical reason for having these additional global brackets could be indeed the configuration of the logger. It needs to be set up before we start with the rest. Just my gut feeling, I did not have a look.

> It would also make an extension to support more SolverInterface objects in a participant more complicated (see https://github.com/precice/precice/issues/655).

No, I think this is not the case. Multiple `SolverInterface` objects would mean multiple `participant` tags are handles by one program. It would only make the name of the tag `solver-interface` even more confusing :see_no_evil:

**Comment by fsimonis** [comment]
The logger is initialized before any configuration parsing occurs.
We need a working logger to allow logs in the configuration parser.

> Who logs while the logger is configured?

**Comment by fsimonis** [comment]
There are two changes in the pipeline that would also be configured outside of the `solver-interface` tag:

* the events overhaul #1410 and #1419
* the unified shared directory #1429

**Comment by fsimonis** [comment]
> Removing the tag altogether and bringing everything to the top level would mean that we need to find a new place not only for the dimensions attribute, but also for the experimental flag. It would also make an extension to support more SolverInterface objects in a participant more complicated (see https://github.com/precice/precice/issues/655).

As a first step, we could move the attributes of the `solver-interface` into the `precice-configuration` tag.

@uekerman `sync-mode` should really be moved to the `profiling` tag introduced in #1410. Do you agree?

The `experimental` attribute makes more sense in the `precice-configuration` tag. We allow the entire config to use experimental features ...

@MakisH Could you open a separate issue regarding the location of the `dimension`?
These affects so many places, that discussions will get quickly out of hand.

**Comment by uekerman** [comment]
> @uekerman sync-mode should really be moved to the profiling tag introduced in https://github.com/precice/precice/issues/1410. Do you agree?

Yes
